org-quick-peek.el

```diff
@@ -144,6 +153,30 @@
         (unless quiet
           (minibuffer-message "Entry has no text.")))))
 
+(cl-defun org-quick-peek--agenda-show-logbook (&key quiet)
+  "Show quick peek (:LOGBOOK: drawer only) of item at current line."
+  (-if-let* ((marker (org-get-at-bol 'org-hd-marker))
+             (text (s-trim
+                    (org-quick-peek--s-trim-lines
+                    (let ((entry 
+                           (org-quick-peek--get-entry-text marker
+                                                           :num-lines nil
+                                                           :keep-drawers t)))
+                      (with-temp-buffer
+                        (setq buffer-read-only nil)
+                        (insert entry)
+                        (beginning-of-buffer)
+                        (when (re-search-forward ":LOGBOOK:" nil)
+                          (let* ((elt (org-element-property-drawer-parser nil))
```

Comment:
Since we're using dash.el, you could use it to destructure the return value of `org-element-property-drawer-parser`.  Or you could use `pcase-let*` to do that as well (see recent update to map.el 2.1 which makes this easy: https://git.savannah.gnu.org/cgit/emacs.git/commit/etc/NEWS?id=e287da5a8154d83a97107b64915ccc17e3a086b8).

Reply:
OK sounds good will take a stab at this

Reply:
It's not necessary to wait for Emacs 28, because that commit was published to ELPA as map.el 2.1, which can be installed on older Emacsen.